pkg/kinds/lookup.go

```diff
@@ -27,11 +28,22 @@ import (
 // Lookup returns the GVK of a object based on the types registered with the
 // provided Scheme.
 func Lookup(obj runtime.Object, scheme *runtime.Scheme) (schema.GroupVersionKind, error) {
-	gvk, err := apiutil.GVKForObject(obj, scheme)
-	if err != nil {
-		return schema.GroupVersionKind{}, fmt.Errorf("failed to lookup object type: %w", err)
+	switch obj.(type) {
+	case *metav1.PartialObjectMetadata:
```

Comment:
Maybe I misread this earlier...
In what case is version not required?

Reply:
discussed offline, moved workaround back to fake client